src/gentropy/dataset/l2g_features/vep.py

```diff
@@ -0,0 +1,268 @@
+    if isinstance(study_loci_to_annotate, StudyLocus):
+        variants_df = (
+            study_loci_to_annotate.df.withColumn(
+                "variantInLocus", f.explode_outer("locus")
+            )
+            .select(
+                "studyLocusId",
+                f.col("variantInLocus.variantId").alias("variantId"),
+                f.col("variantInLocus.posteriorProbability").alias(
+                    "posteriorProbability"
+                ),
+            )
+            .join(consequences_dataset, "variantId")
+        )
+    elif isinstance(study_loci_to_annotate, L2GGoldStandard):
+        variants_df = study_loci_to_annotate.df.select(
+            "studyLocusId", "variantId", f.lit(1.0).alias("posteriorProbability")
+        ).join(consequences_dataset, "variantId")
+
+    if "Maximum" in feature_name:
+        agg_expr = f.max("severityScore")
+    elif "Mean" in feature_name:
+        variants_df = variants_df.withColumn(
+            "weightedScore", f.col("severityScore") * f.col("posteriorProbability")
+        )
+        agg_expr = f.mean("weightedScore")
+    return variants_df.groupBy("studyLocusId", "geneId").agg(
+        agg_expr.alias(feature_name)
+    )
+
+
+def common_neighbourhood_vep_feature_logic(
+    study_loci_to_annotate: StudyLocus | L2GGoldStandard,
+    *,
+    variant_index: VariantIndex,
+    gene_index: GeneIndex,
+    feature_name: str,
+) -> DataFrame:
+    """Extracts variant severity score computed from VEP for any gene, based on what is the mean score for protein coding genes that are nearby the locus.
+
+    Args:
+        study_loci_to_annotate (StudyLocus | L2GGoldStandard): The dataset containing study loci that will be used for annotation
+        variant_index (VariantIndex): The dataset containing functional consequence information
+        gene_index (GeneIndex): The dataset containing the gene biotype
+        feature_name (str): The name of the feature
+
+    Returns:
+        DataFrame: Feature dataset
+    """
+    local_feature_name = feature_name.replace("Neighbourhood", "")
+    # First compute mean distances to a gene
+    local_metric = common_vep_feature_logic(
+        study_loci_to_annotate,
+        feature_name=local_feature_name,
+        variant_index=variant_index,
+    )
+    return (
+        # Then compute mean distance in the vicinity (feature will be the same for any gene associated with a studyLocus)
+        local_metric.join(
+            # Bring gene classification
+            gene_index.df.select("geneId", "biotype"),
+            "geneId",
+            "inner",
+        )
+        .withColumn(
+            "regional_metric",
+            f.coalesce(
+                # Calculate mean based on protein coding genes
+                f.mean(
+                    f.when(
+                        f.col("biotype") == "protein_coding", f.col(local_feature_name)
+                    )
+                ).over(Window.partitionBy("studyLocusId")),
```

Comment:
How do we assign the genes for this studyLocusId?

Reply:
Genes come from VEP. It's the genes of the transcript that the variant overlaps with in a 500kb window

Reply:
For variantId - yes, but we have many varintIds in one studyLocusId. So just to be sure - we use the union across al varaintIds? And if varintId is not assigned to geneId we assume its VEP score is 0?